app/controllers/concerns/git_interface.rb

```diff
@@ -2,69 +2,101 @@ module GitInterface
-    from_branch = id
-    their_commit = repo.branches[into_branch].target_id
-    our_commit = repo.branches[from_branch].target_id
-
-    merge_index = repo.merge_commits(our_commit, their_commit)
-
-    if merge_index.conflicts?
-      # conflicts. deal with them
-    else
-      # no conflicts
-      commit_tree = merge_index.write_tree(repo)
+  def rugged_create (id, branch_id, string, action)
+    begin
+      repo = setup
+      #find/create branch and check it out
+      branch = repo.branches[branch_id]
+      if branch.nil?
+        branch = repo.branches.create(branch_id, "HEAD")
+      end
+      repo.checkout(branch)
+      #add blob
+      oid = repo.write(string,:blob)
+      index = repo.index
+      index.read_tree(repo.head.target.tree)
+      index.add(:path => "#{id}.nt", :oid => oid, :mode => 0100644)
+      #commit
       options = {}
-      options[:tree] = commit_tree
-      options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
-      options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
-      options[:message] ||= "Merge #{from_branch} into #{into_branch}"
-      options[:parents] = [repo.head.target, our_commit]
+      options[:tree] = index.write_tree(repo)
+      options[:author] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
+      options[:committer] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
+      options[:message] = action + ": " + id
+      options[:parents] = repo.empty? ? [] : [ repo.head.target ].compact
       options[:update_ref] = 'HEAD'
-
       Rugged::Commit.create(repo, options)
-      repo.checkout_tree(commit_tree)
-      index = repo.index
       index.write
       options = {}
       options[:strategy] = :force
       repo.checkout_head(options)
-      #repo.push('origin', [repo.head.name], { credentials: @cred })
-      repo.branches.delete(from_branch)
-      #close branch
+      repo.checkout('master')
+    rescue
+      logger.error = ("Git create failed. Refer to " + branch_id)
+    end
+  end
+
+  def rugged_merge (id, branch_id)
+    begin
+      repo = setup
+      repo.checkout('master')
+      #merge
+      into_branch = 'master'
+      from_branch = branch_id
+      their_commit = repo.branches[into_branch].target_id
+      our_commit = repo.branches[from_branch].target_id
+
+      merge_index = repo.merge_commits(our_commit, their_commit)
+
+      if merge_index.conflicts?
+        logger.error('merge conflict with branch: ' + branch_id)
+        raise 'merge conflict'
+      else
+        commit_tree = merge_index.write_tree(repo)
+        options = {}
+        options[:tree] = commit_tree
+        options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
+        options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
```

Comment:
These few lines keep reoccurring. Maybe we could split it into its own method to cut down on repetition? I see something like
```
def set_author_committer
     options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
     options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
end
```